### PR TITLE
feat(active-toggle): adding toggle button for user activity status

### PR DESCRIPTION
**Pull request by OrelKaravani**
getting the activity status from DB and showing in the toggle button
saving status when user switch the button to 'Active' or 'Inactive'

### CODE REVIEW

File: client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                customClass: {
+                    title: classes.swalTitle
+                },
+            });
+        });
+    }
+
+    const setUserActivityStatus = (userId: string, isActive: boolean) => {
+        axios.post('usersInfo/updateIsUserActive', {
+            id: userId,
```
Comment (sandymash) [inline client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx:37]: dont send user id implicitly through a request

Comment (sandymash) [reply client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx:37]: the user id is updated automatically according to the token in the server, you can access it through _response.locals_ as it is done in other cases

File: client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+export interface useTopToolbarOutcome  {
+    getUserActivityStatus: (userId: string) => void;
+    setUserActivityStatus: (userId: string, isActive: boolean) => void;
+}
+
+const useTopToolbar = (props: useInteractionsTabInput) :  useTopToolbarOutcome => {
+
+    const classes = useStyles({});
+
+    const getUserActivityStatus = (userId: string) => {
```
Comment (sandymash) [inline client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx:20]: same here about the userID

File: client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx (hunk 1 above)
Comment (OrelKaravani) [reply client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx]: Fixed

File: client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx (hunk 2 above)
Comment (OrelKaravani) [reply client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx:20]: Fixed

**Review by sandymash — DISMISSED** [review]

File: client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx [hunk 3]
```diff
@@ -2,21 +2,71 @@ import React from 'react';
...
             <div className={classes.rightToolbarSection}>
                 <img alt='logo' className={classes.logo} src='./assets/img/logo.png'></img>
                 <Typography variant='h4' className={classes.centering}><div className={classes.systemName}>אבן יסוד</div></Typography>
             </div>
-            <Typography>שלום, {userName}</Typography>
+            <div className={classes.leftToolbarSection}>
+                <Tooltip title={toggleMessage} arrow>
+                    <ToggleButtonGroup value={isActive} exclusive className={classes.isActiveToggle}>
+                        <ToggleButton className={classes.toggle} 
+                            style={isActive ? activeButtonStyle : {}}
```
Comment (MayaSasson) [inline client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx:55]: why style and not class? same for line 61

Comment (OrelKaravani) [reply client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx:55]: This is how the common toggle implemented too.. Class doesn't work here. I would like to know how do you suggest to do it.

File: client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    setIsActive: (isActive: boolean) => void;
+};
+
+
+export interface useTopToolbarOutcome  {
+    getUserActivityStatus: () => void;
+    setUserActivityStatus: (isActive: boolean) => void;
+}
+
+const useTopToolbar = (props: useInteractionsTabInput) :  useTopToolbarOutcome => {
```
Comment (RoeiRom) [inline client/src/components/App/AppToolbar/TopToolbar/useTopToolbar.tsx:16]: destruct the props, do { blabla }: useBlaBla

File: client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx (hunk 3 above)
Comment (sandymash) [reply client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx:55]: a classes sub-object returns a string you could add it conditionally 🤷‍♀️

**Review by sandymash — DISMISSED** [review]

File: server/src/ClientToDBAPI/mainRoute.ts [hunk 5]
```diff
@@ -18,5 +19,6 @@ clientToDBRouter.use('/investigationInfo', investigationInfo);
 clientToDBRouter.use('/personalDetails', personalDetailsRoute);
 clientToDBRouter.use('/clinicalDetails', clinicalDetailsRoute);
 clientToDBRouter.use('/addressDetails', addressRoute);
+clientToDBRouter.use('/usersInfo', usersRoute);
```
Comment (RoeiRom) [inline server/src/ClientToDBAPI/mainRoute.ts:22]: change route to /users

File: client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx [hunk 6]
```diff
@@ -2,21 +2,71 @@ import React from 'react';
...
+
+    const notActiveButtonStyle = {
+        backgroundColor: '#9e2a2b',
+        color: 'white'
+    }
+
+    React.useEffect( () => {
+        if (firstUserUpdate.current) {
+            firstUserUpdate.current = false;
+            return;
```
Comment (RoeiRom) [inline client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx:39]: dont need the return

**Review by RoeiRom — CHANGES_REQUESTED** [review]

File: client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx (hunk 3 above)
Comment (OrelKaravani) [reply client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx:54]: I tried. Didn't work for me. If you did it I would like to see how you made it happen.

**Review by RoeiRom — DISMISSED** [review]

File: client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx (hunk 3 above)
Comment (sandymash) [reply client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx:54]:
```suggestion
                            className={`${classes.toggle} ${isActive ? classes.activeButtonStyle:''}`} 
```

Comment (sandymash) [reply client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx:54]:
```suggestion
							const toggleClasses = isActive :[classes.toggle, classes.activeButtonStyle ] : [classes.toggle] 
                            className={toggleClasses.join(' ')} 
```

Comment (sandymash) [reply client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx:54]: there are probably better ways than these but those give out a general idea

Comment (OrelKaravani) [reply client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx]: Doesn't work both ways.

**Review by RoeiRom — APPROVED** [review]

**Review by EmilyLip — APPROVED** [review]

**Pull request merged**